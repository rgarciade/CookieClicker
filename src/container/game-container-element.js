import { LitElement, html } from 'lit';
import '../components/button-element.js';
import { gameStyle } from './styles/game-container-style.js';
import { commonStyle } from '../styles/common.js';
import ClickerService from '../services/clicker/clicker-service.js';
import clickerTypes from '../services/clicker/clicker-types.js';
import '../components/nav-element.js';
import { ScoresState } from '../status/index.js';

export class GameContainerElement extends ScoresState(LitElement) {
	static get styles() {
		return [gameStyle, commonStyle];
	}
	/**
	 * @property {String} name - The name of the gamer.
	 */
	static get properties() {
		return {
			name: { type: String },
		};
	}
	_initPlayerData() {
		if (this.name == '') return;
		let playerData = this._scores_getScoreByName(this.name);
		this.clicker.initialize({
			initialValue: playerData.value,
			initialNumberOfClickers: playerData.numberOfClickers,
		});
	}
	_updateClickerActualValue() {
		const oldVal = this.counterValue;
		this.counterValue = this.clicker.actualValue;
		this.requestUpdate('counterValue', oldVal);
		if (oldVal !== this.counterValue) {
			this._scores_updateScore(this.name, {
				value: this.clicker.actualValue,
				numberOfClickers: this.clicker.numberOfClickers,
			});
		}
	}
	_updateClickerActualClickerPrice() {
		const oldVal = this.actualClickerPrice;
		this.actualClickerPrice = this.clicker.nextClickerPrice(
			clickerTypes.BasicAutoClicker
		);
		this.requestUpdate('actualClickerPrice', oldVal);
	}
	_updateClickerBasicAutoClickerEnabled() {
		const oldVal = this.basicAutoClickerEnabled;
		this.basicAutoClickerEnabled = this.clicker.checkClickerEnabled(
			clickerTypes.BasicAutoClicker
		);
		this.requestUpdate('basicAutoClickerEnabled', oldVal);
	}

	_updateClickerMegaClickersActualClickerPrice() {
		const oldVal = this.megaClickersActualClickerPrice;
		this.megaClickersActualClickerPrice = this.clicker.nextClickerPrice(
			clickerTypes.MegaClickers
		);
		this.requestUpdate('megaClickersActualClickerPrice', oldVal);
	}
	_updateClickerMegaClickersEnabled() {
		const oldVal = this.megaClickersEnabled;
		this.megaClickersEnabled = this.clicker.checkClickerEnabled(
			clickerTypes.MegaClickers
		);
		this.requestUpdate('megaClickersEnabled', oldVal);
	}

	constructor() {
		super();
		this.actualClickerPrice = 0;
		this.counterValue = 0;
		this.basicAutoClickerEnabled = false;

		this.megaClickersActualClickerPrice = 0;
		this.megaClickersEnabled = false;

		this.clicker = new ClickerService({
			updateStatusFunction: () => {
				this._updateClickerActualValue();
				this._updateClickerActualClickerPrice();
				this._updateClickerBasicAutoClickerEnabled();
				this._updateClickerMegaClickersActualClickerPrice();
				this._updateClickerMegaClickersEnabled();
			},
		});
	}
	disconnectedCallback() {
		// es importante destruir el servicio clicker
		//  para que no siga con el setInterval
		this.clicker.dismount();
		super.disconnectedCallback();
	}
	update(changedProperties) {
		super.update(changedProperties);
		if (!changedProperties.has('name')) return;
		this._initPlayerData();
	}

	addOne() {
		this.clicker.buyClicker(clickerTypes.SimpleClicker);
	}
	buyClicker() {
		this.clicker.buyClicker(clickerTypes.BasicAutoClicker);
	}
	buyMegaClicker() {
		this.clicker.buyClicker(clickerTypes.MegaClickers);
	}
	render() {
		return html`
			<nav-element href="/identify">change player</nav-element>

			<div class="container">
				<p class="game-text">
					Merge Pull Request: <span>${this.counterValue}</span>
				</p>
				<button-element @click="${this.addOne}" width="10em">
					<icon-element icon="merge_type" size="50px"></icon-element>
					Add 1
				</button-element>
				<div class="auto-clicker-group">
					<button-element
						?disabled="${!this.basicAutoClickerEnabled}"
						@click="${this.buyClicker}"
						width="8.5em"
						fontSize="1em"
					>
						<icon-element icon="merge_type" size="50px"></icon-element>
						Add clicker: ${this.actualClickerPrice}
					</button-element>
					<button-element
						?disabled="${!this.megaClickersEnabled}"
						@click="${this.buyMegaClicker}"
						width="10em"
						fontSize="1em"
					>
						<icon-element icon="merge_type" size="50px"></icon-element>
						Add Mega clicker: ${this.megaClickersActualClickerPrice}
					</button-element>
				</div>
			</div>
		`;
	}
}

customElements.define('game-container-element', GameContainerElement);
