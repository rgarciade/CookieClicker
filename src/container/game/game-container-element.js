import { LitElement, html } from 'lit';
import '../../components/button/button-element.js';
import { gameStyle } from './game-container-style.js';
import { commonStyle } from '../../styles/common.js';
import ClickerService from '../../services/clicker/clicker-service.js';
import clickerTypes from '../../services/clicker/clicker-types.js';
import '../../components/nav/nav-element.js';
import '../../components/gameTitle/game-title-element.js';
import { ScoresState } from '../../status/index.js';
import { navigate } from '../../services/navigate/navigate-service.js';

//ScoresState es un mixin que nos permite acceder a los scores del juego y modificarlos
export class GameContainerElement extends ScoresState(LitElement) {
	static get styles() {
		return [gameStyle, commonStyle];
	}
	/**
	 * @property {String} name - The name of the gamer.
	 */
	static get properties() {
		return {
			name: { type: String },
		};
	}
	_initPlayerData() {
		if (this.name == '') return;
		let playerData = this._scores_getScoreByName(this.name);
		//para que el clicker funcione necesita inicializarse cuando se tiene el nombre del jugador
		this.clicker.initialize({
			initialValue: playerData.value,
			initialNumberOfClickers: playerData.numberOfClickers,
		});
	}
	_updateClickerActualValue() {
		//usamos requestUpdate para que se actualice el valor en el html
		const oldVal = this.counterValue;
		this.counterValue = this.clicker.actualValue;
		this.requestUpdate('counterValue', oldVal);
		if (oldVal !== this.counterValue) {
			this._scores_updateScore(this.name, {
				value: this.clicker.actualValue,
				numberOfClickers: this.clicker.numberOfClickers,
			});
		}
	}
	_updateClickerActualClickerPrice() {
		const oldVal = this.actualClickerPrice;
		this.actualClickerPrice = this.clicker.nextClickerPrice(
			clickerTypes.BasicAutoClicker
		);
		this.requestUpdate('actualClickerPrice', oldVal);
	}
	_updateClickerBasicAutoClickerEnabled() {
		const oldVal = this.basicAutoClickerEnabled;
		this.basicAutoClickerEnabled = this.clicker.checkClickerEnabled(
			clickerTypes.BasicAutoClicker
		);
		this.requestUpdate('basicAutoClickerEnabled', oldVal);
	}

	_updateClickerMegaClickersActualClickerPrice() {
		const oldVal = this.megaClickersActualClickerPrice;
		this.megaClickersActualClickerPrice = this.clicker.nextClickerPrice(
			clickerTypes.MegaClickers
		);
		this.requestUpdate('megaClickersActualClickerPrice', oldVal);
	}
	_updateClickerMegaClickersEnabled() {
		const oldVal = this.megaClickersEnabled;
		this.megaClickersEnabled = this.clicker.checkClickerEnabled(
			clickerTypes.MegaClickers
		);
		this.requestUpdate('megaClickersEnabled', oldVal);
	}
	_updateNumberOfClickers() {
		const oldVal = this.clicker.numberOfClickers;
		this.numberOfClickers = this.clicker.numberOfClickers;
		this.requestUpdate('numberOfClickers', oldVal);
	}

	constructor() {
		super();
		this.actualClickerPrice = 0;
		this.counterValue = 0;
		this.numberOfClickers = 0;
		this.basicAutoClickerEnabled = false;

		this.megaClickersActualClickerPrice = 0;
		this.megaClickersEnabled = false;

		//definimos el servicio clicker sin inicializarlo, ya que no tenemos el nombre del jugador
		this.clicker = new ClickerService({
			updateStatusFunction: () => {
				//actualizamos todos los valores modificables del componente
				this._updateClickerActualValue();
				this._updateClickerActualClickerPrice();
				this._updateClickerBasicAutoClickerEnabled();
				this._updateClickerMegaClickersActualClickerPrice();
				this._updateClickerMegaClickersEnabled();
				this._updateNumberOfClickers();
			},
		});
	}
	disconnectedCallback() {
		// es importante destruir el servicio clicker
		//  para que no siga con el setInterval
		this.clicker.dismount();
		super.disconnectedCallback();
	}
	update(changedProperties) {
		super.update(changedProperties);
		if (!changedProperties.has('name')) return;
		this._initPlayerData();
	}

	addOne() {
		this.clicker.buyClicker(clickerTypes.SimpleClicker);
	}
	buyClicker() {
		this.clicker.buyClicker(clickerTypes.BasicAutoClicker);
	}
	buyMegaClicker() {
		this.clicker.buyClicker(clickerTypes.MegaClickers);
	}
	navegate(path) {
		// en identify-container e usado otra forma de navegar, para probar varios conceptos
		navigate(this, path);
	}
	render() {
		// no e visto necesario dividir en más componentes la vista, si creciera y viera que se pueden reutilizar partes, crearía más componentes
		return html`
			<game-title-element class="game-title">
				<icon-element icon="person" class="person"></icon-element>
				<p>${this.name}</p>
				<icon-element
					class="go-identify"
					icon="logout"
					@click="${() => this.navegate('/identify')}"
				></icon-element>
			</game-title-element>
			<div class="container">
				<div class="game-text">
					<p>Merge Pull Request: <span>${this.counterValue}</span></p>
					<p>Auto mergers: ${this.numberOfClickers}</p>
				</div>

				<button-element @click="${this.addOne}" class="individual-clicker">
					<icon-element icon="merge_type"></icon-element>
					Add 1
				</button-element>
				<div class="auto-clicker-group">
					<button-element
						class="small-clicker"
						?hidden="${!this.basicAutoClickerEnabled}"
						@click="${this.buyClicker}"
					>
						<icon-element icon="merge_type"></icon-element>
						Add clicker: ${this.actualClickerPrice}
					</button-element>
					<button-element
						class="big-clicker"
						?hidden="${!this.megaClickersEnabled}"
						@click="${this.buyMegaClicker}"
					>
						<icon-element icon="merge_type"></icon-element>
						Add Mega clicker: ${this.megaClickersActualClickerPrice}
					</button-element>
				</div>
			</div>
		`;
	}
}

customElements.define('game-container-element', GameContainerElement);
