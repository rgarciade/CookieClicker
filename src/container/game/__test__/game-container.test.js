import { html, fixture } from '@open-wc/testing';
import '../game-container-element.js';
import { expect } from '@jest/globals';
import { cleanedLitElements } from '../../../../testFunctions/cleanedLit.js';
import { getAllElementsAndShadowRoots } from 'shadow-dom-testing-library';

describe('game-container-element', () => {
	it('game-container matches the snapshot', async () => {
		const el = await fixture(
			html`<game-container-element name="nombre test"></game-container-element>`
		);
		const data = getAllElementsAndShadowRoots(el, '*');
		const dataCleaned = cleanedLitElements(data);
		expect(dataCleaned).toMatchSnapshot();
	});
});
