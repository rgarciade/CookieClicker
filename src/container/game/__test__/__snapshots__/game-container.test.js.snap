// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`game-container-element game-container matches the snapshot 1`] = `
[
  "[36m<game-container-element[39m
  [33mname[39m=[32m"nombre test"[39m
[36m/>[39m",
  "ShadowRoot {}",
  "ShadowRoot {}",
  "[36m<slot />[39m",
  "[36m<style>[39m
  [0m
	:host {
		font-size: var(--title-font-size, 70px);
		color: var(--text-color, black);
		display: flex;
		gap: 40px;
		justify-content: flex-start;

		background-color: var(--background-color, white);
	}
	.game-title :last-child {
		margin-left: auto;
		padding-right: 20px;
	}
	.game-title :first-child {
		padding-left: 20px;
	}

	:host ::slotted(*) {
		display: flex;
		justify-content: flex-start;
		align-items: center;
		line-height: 0.2;
	}

	:host ::slotted(*:last-child) {
		margin-left: auto;
		padding-right: 40px;
	}
	:host ::slotted(*:first-child) {
		padding-left: 40px;
	}
[0m
[36m</style>[39m",
  "ShadowRoot {}",
  "[36m<i[39m
  [33mclass[39m=[32m"material-icons"[39m
[36m>[39m
  [0mperson[0m
[36m</i>[39m",
  "[36m<style>[39m
  [0m
	:host {
		--mdc-icon-size: 24px;
	}
	.material-icons {
		font-family: 'Material Icons';
		font-weight: normal;
		font-style: normal;
		font-size: var(--icon-size, 120px); /* Preferred icon size */
		display: inline-block;
		line-height: 1;
		text-transform: none;
		letter-spacing: normal;
		word-wrap: normal;
		white-space: nowrap;
		direction: ltr;
		color: var(--icon-color, white);

		/* Support for all WebKit browsers. */
		-webkit-font-smoothing: antialiased;
		/* Support for Safari and Chrome. */
		text-rendering: optimizeLegibility;

		/* Support for Firefox. */
		-moz-osx-font-smoothing: grayscale;

		/* Support for IE. */
		font-feature-settings: 'liga';
	}
[0m
[36m</style>[39m",
  "[36m<p>[39m
  [0mnombre test[0m
[36m</p>[39m",
  "ShadowRoot {}",
  "[36m<i[39m
  [33mclass[39m=[32m"material-icons"[39m
[36m>[39m
  [0mlogout[0m
[36m</i>[39m",
  "[36m<style>[39m
  [0m
	:host {
		--mdc-icon-size: 24px;
	}
	.material-icons {
		font-family: 'Material Icons';
		font-weight: normal;
		font-style: normal;
		font-size: var(--icon-size, 120px); /* Preferred icon size */
		display: inline-block;
		line-height: 1;
		text-transform: none;
		letter-spacing: normal;
		word-wrap: normal;
		white-space: nowrap;
		direction: ltr;
		color: var(--icon-color, white);

		/* Support for all WebKit browsers. */
		-webkit-font-smoothing: antialiased;
		/* Support for Safari and Chrome. */
		text-rendering: optimizeLegibility;

		/* Support for Firefox. */
		-moz-osx-font-smoothing: grayscale;

		/* Support for IE. */
		font-feature-settings: 'liga';
	}
[0m
[36m</style>[39m",
  "[36m<div[39m
  [33mclass[39m=[32m"container"[39m
[36m>[39m
  [0m
				[0m
  [36m<div[39m
    [33mclass[39m=[32m"game-text"[39m
  [36m>[39m
    [0m
					[0m
    [36m<p>[39m
      [0mMerge Pull Request: [0m
      [36m<span>[39m
        [0m0[0m
      [36m</span>[39m
    [36m</p>[39m
    [0m
					[0m
    [36m<p>[39m
      [0mAuto mergers: [0m
      [0m0[0m
    [36m</p>[39m
    [0m
				[0m
  [36m</div>[39m
  [0m

				[0m
  [36m<button-element[39m
    [33mclass[39m=[32m"individual-clicker"[39m
  [36m>[39m
    [0m
					[0m
    [36m<icon-element[39m
      [33micon[39m=[32m"merge_type"[39m
    [36m/>[39m
    [0m
					Add 1
				[0m
  [36m</button-element>[39m
  [0m
				[0m
  [36m<div[39m
    [33mclass[39m=[32m"auto-clicker-group"[39m
  [36m>[39m
    [0m
					[0m
    [36m<button-element[39m
      [33mclass[39m=[32m"small-clicker"[39m
      [33mhidden[39m=[32m""[39m
    [36m>[39m
      [0m
						[0m
      [36m<icon-element[39m
        [33micon[39m=[32m"merge_type"[39m
      [36m/>[39m
      [0m
						Add clicker: [0m
      [0m50[0m
      [0m
					[0m
    [36m</button-element>[39m
    [0m
					[0m
    [36m<button-element[39m
      [33mclass[39m=[32m"big-clicker"[39m
      [33mhidden[39m=[32m""[39m
    [36m>[39m
      [0m
						[0m
      [36m<icon-element[39m
        [33micon[39m=[32m"merge_type"[39m
      [36m/>[39m
      [0m
						Add Mega clicker: [0m
      [0m1000[0m
      [0m
					[0m
    [36m</button-element>[39m
    [0m
				[0m
  [36m</div>[39m
  [0m
			[0m
[36m</div>[39m",
  "[36m<div[39m
  [33mclass[39m=[32m"game-text"[39m
[36m>[39m
  [0m
					[0m
  [36m<p>[39m
    [0mMerge Pull Request: [0m
    [36m<span>[39m
      [0m0[0m
    [36m</span>[39m
  [36m</p>[39m
  [0m
					[0m
  [36m<p>[39m
    [0mAuto mergers: [0m
    [0m0[0m
  [36m</p>[39m
  [0m
				[0m
[36m</div>[39m",
  "[36m<p>[39m
  [0mMerge Pull Request: [0m
  [36m<span>[39m
    [0m0[0m
  [36m</span>[39m
[36m</p>[39m",
  "[36m<span>[39m
  [0m0[0m
[36m</span>[39m",
  "[36m<p>[39m
  [0mAuto mergers: [0m
  [0m0[0m
[36m</p>[39m",
  "ShadowRoot {}",
  "[36m<button[39m
  [33mclass[39m=[32m"primary long"[39m
[36m>[39m
  [0m
				[0m
  [36m<slot />[39m
  [0m
			[0m
[36m</button>[39m",
  "[36m<slot />[39m",
  "[36m<style>[39m
  [0m
	button {
		border-radius: 12px;
		background-color: var(--color-primario);
		border: none;
		color: white;
		padding: 15px 32px;
		text-align: center;
		text-decoration: none;
		display: inline-block;
		font-size: var(--button-font-size, 76px);
		margin: 4px 2px;
		cursor: pointer;
		width: var(--button-width, '2px');
		min-width: var(--button-width, '2px');
		height: var(--button-height, '2px');
	}
	button:disabled {
		background-color: grey;
	}
	.primary {
		background-color: var(--color-primario, blue);
	}
	.secondary {
		background-color: var(--color-secundario, red);
	}
	.long {
		width: 100%;
	}
	.short {
		width: 50%;
	}
[0m
[36m</style>[39m",
  "ShadowRoot {}",
  "[36m<i[39m
  [33mclass[39m=[32m"material-icons"[39m
[36m>[39m
  [0mmerge_type[0m
[36m</i>[39m",
  "[36m<style>[39m
  [0m
	:host {
		--mdc-icon-size: 24px;
	}
	.material-icons {
		font-family: 'Material Icons';
		font-weight: normal;
		font-style: normal;
		font-size: var(--icon-size, 120px); /* Preferred icon size */
		display: inline-block;
		line-height: 1;
		text-transform: none;
		letter-spacing: normal;
		word-wrap: normal;
		white-space: nowrap;
		direction: ltr;
		color: var(--icon-color, white);

		/* Support for all WebKit browsers. */
		-webkit-font-smoothing: antialiased;
		/* Support for Safari and Chrome. */
		text-rendering: optimizeLegibility;

		/* Support for Firefox. */
		-moz-osx-font-smoothing: grayscale;

		/* Support for IE. */
		font-feature-settings: 'liga';
	}
[0m
[36m</style>[39m",
  "[36m<div[39m
  [33mclass[39m=[32m"auto-clicker-group"[39m
[36m>[39m
  [0m
					[0m
  [36m<button-element[39m
    [33mclass[39m=[32m"small-clicker"[39m
    [33mhidden[39m=[32m""[39m
  [36m>[39m
    [0m
						[0m
    [36m<icon-element[39m
      [33micon[39m=[32m"merge_type"[39m
    [36m/>[39m
    [0m
						Add clicker: [0m
    [0m50[0m
    [0m
					[0m
  [36m</button-element>[39m
  [0m
					[0m
  [36m<button-element[39m
    [33mclass[39m=[32m"big-clicker"[39m
    [33mhidden[39m=[32m""[39m
  [36m>[39m
    [0m
						[0m
    [36m<icon-element[39m
      [33micon[39m=[32m"merge_type"[39m
    [36m/>[39m
    [0m
						Add Mega clicker: [0m
    [0m1000[0m
    [0m
					[0m
  [36m</button-element>[39m
  [0m
				[0m
[36m</div>[39m",
  "ShadowRoot {}",
  "[36m<button[39m
  [33mclass[39m=[32m"primary long"[39m
  [33mhidden[39m=[32m""[39m
[36m>[39m
  [0m
				[0m
  [36m<slot />[39m
  [0m
			[0m
[36m</button>[39m",
  "[36m<slot />[39m",
  "[36m<style>[39m
  [0m
	button {
		border-radius: 12px;
		background-color: var(--color-primario);
		border: none;
		color: white;
		padding: 15px 32px;
		text-align: center;
		text-decoration: none;
		display: inline-block;
		font-size: var(--button-font-size, 76px);
		margin: 4px 2px;
		cursor: pointer;
		width: var(--button-width, '2px');
		min-width: var(--button-width, '2px');
		height: var(--button-height, '2px');
	}
	button:disabled {
		background-color: grey;
	}
	.primary {
		background-color: var(--color-primario, blue);
	}
	.secondary {
		background-color: var(--color-secundario, red);
	}
	.long {
		width: 100%;
	}
	.short {
		width: 50%;
	}
[0m
[36m</style>[39m",
  "ShadowRoot {}",
  "[36m<i[39m
  [33mclass[39m=[32m"material-icons"[39m
[36m>[39m
  [0mmerge_type[0m
[36m</i>[39m",
  "[36m<style>[39m
  [0m
	:host {
		--mdc-icon-size: 24px;
	}
	.material-icons {
		font-family: 'Material Icons';
		font-weight: normal;
		font-style: normal;
		font-size: var(--icon-size, 120px); /* Preferred icon size */
		display: inline-block;
		line-height: 1;
		text-transform: none;
		letter-spacing: normal;
		word-wrap: normal;
		white-space: nowrap;
		direction: ltr;
		color: var(--icon-color, white);

		/* Support for all WebKit browsers. */
		-webkit-font-smoothing: antialiased;
		/* Support for Safari and Chrome. */
		text-rendering: optimizeLegibility;

		/* Support for Firefox. */
		-moz-osx-font-smoothing: grayscale;

		/* Support for IE. */
		font-feature-settings: 'liga';
	}
[0m
[36m</style>[39m",
  "ShadowRoot {}",
  "[36m<button[39m
  [33mclass[39m=[32m"primary long"[39m
  [33mhidden[39m=[32m""[39m
[36m>[39m
  [0m
				[0m
  [36m<slot />[39m
  [0m
			[0m
[36m</button>[39m",
  "[36m<slot />[39m",
  "[36m<style>[39m
  [0m
	button {
		border-radius: 12px;
		background-color: var(--color-primario);
		border: none;
		color: white;
		padding: 15px 32px;
		text-align: center;
		text-decoration: none;
		display: inline-block;
		font-size: var(--button-font-size, 76px);
		margin: 4px 2px;
		cursor: pointer;
		width: var(--button-width, '2px');
		min-width: var(--button-width, '2px');
		height: var(--button-height, '2px');
	}
	button:disabled {
		background-color: grey;
	}
	.primary {
		background-color: var(--color-primario, blue);
	}
	.secondary {
		background-color: var(--color-secundario, red);
	}
	.long {
		width: 100%;
	}
	.short {
		width: 50%;
	}
[0m
[36m</style>[39m",
  "ShadowRoot {}",
  "[36m<i[39m
  [33mclass[39m=[32m"material-icons"[39m
[36m>[39m
  [0mmerge_type[0m
[36m</i>[39m",
  "[36m<style>[39m
  [0m
	:host {
		--mdc-icon-size: 24px;
	}
	.material-icons {
		font-family: 'Material Icons';
		font-weight: normal;
		font-style: normal;
		font-size: var(--icon-size, 120px); /* Preferred icon size */
		display: inline-block;
		line-height: 1;
		text-transform: none;
		letter-spacing: normal;
		word-wrap: normal;
		white-space: nowrap;
		direction: ltr;
		color: var(--icon-color, white);

		/* Support for all WebKit browsers. */
		-webkit-font-smoothing: antialiased;
		/* Support for Safari and Chrome. */
		text-rendering: optimizeLegibility;

		/* Support for Firefox. */
		-moz-osx-font-smoothing: grayscale;

		/* Support for IE. */
		font-feature-settings: 'liga';
	}
[0m
[36m</style>[39m",
  "[36m<style>[39m
  [0m
	.container {
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
		height: 55vh;
		font-size: 40px;
		gap: 20px;
	}
	.auto-clicker-group {
		display: flex;
		gap: 20px;
		max-width: 19em;
		min-height: 6em;
	}
	.game-text {
		font-size: 60px;
		margin: 100px;
		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: center;
	}
	.game-text p {
		margin-bottom: 5px;
		margin-top: 5px;
	}
	.go-identify {
		cursor: pointer;
	}
	.small-clicker {
		--button-height: 5.6em;
		--button-font-size: 40px;
		--icon-size: 70px;
	}
	.big-clicker {
		--button-height: 5.6em;
		--button-font-size: 40px;
		--icon-size: 70px;
	}
	.game-title {
		--background-color: #262626;
	}

	@media only screen and (max-width: 1080px) {
		.game-title {
			--icon-size: 60px;
			--title-font-size: 70px;
		}
		.game-text {
			font-size: 42px;
		}
		.individual-clicker {
			--button-font-size: 50px;
			--icon-size: 60px;
		}
	}
[0m
[36m</style>[39m",
  "[36m<style>[39m
  [0m
	input {
		font-size: 20px;
		padding: 10px;
		border-radius: 5px;
		width: 70px;
	}
	input {
		font-size: 56px;
		padding: var(--spacing, 20px);
		border-radius: 12px;
		border: 1px solid black;
	}
	h1 {
		font-size: 56px;
	}
	select {
		font-size: 36px;
	}
[0m
[36m</style>[39m",
]
`;
